include/NeoFOAM/core/span.hpp

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2025 NeoFOAM authors
+#pragma once
+
+#include <span>
+
+namespace NeoFOAM
+{
+
+template<typename ValueType>
+class Span : public std::span<ValueType>
+{
+public:
+
+    bool abort = true;
+
+    using std::span<ValueType>::span; // Inherit constructors from std::span
+
+    Span(std::span<ValueType> in) : Span(in.begin(), in.end()) {}
+
+    constexpr ValueType& operator[](std::size_t index) const
+    {
+#ifdef NF_DEBUG
+        if (index >= this->size())
+        {
+            std::string msg;
+            msg += "Index is out of range. Index: " + std::to_string(index);
```

Comment:
```suggestion
            std::string msg("Index is out of range. Index: " + std::to_string(index));
```

#pedanticoptimization

Reply:
Interestingly it doesn't work on all compilers 

```
/share/software/actions-runner/_work/NeoFOAM/NeoFOAM/include/NeoFOAM/core/span.hpp:39:25: error: variable of non-literal type 'std::string' (aka 'basic_string<char>') cannot be defined in a constexpr function before C++2b
            std::string msg("Index is out of range. Index: " + std::to_string(index));
                        ^
```